exchange/impcustomcachekeytest/multiImpCustomCacheKey.json

```diff
@@ -0,0 +1,95 @@
+{
+	"bidRequest": {
+		"imp": [{
+				"id": "oneImp",
+				"exp": 300
+			},
+			{
+				"id": "twoImp"
+			}
+		]
+	},
+	"pbsBids": [{
+		"bid": {
+			"cat": ["11_sports_22"],
+			"id": "bidOne",
+			"impid": "oneImp",
+			"price": 7.64,
+			"exp": 600
+		},
+		"bidType": "banner",
+		"bidder": "appnexus"
+	}, {
+		"bid": {
+			"cat": ["33_news_44"],
+			"id": "bidTwo",
+			"impid": "oneImp",
+			"price": 15.64,
+			"exp": 200
+		},
+		"bidType": "banner",
+		"bidder": "pubmatic"
+	}, {
+		"bid": {
+			"cat": ["33_news_44"],
+			"id": "bidThree",
+			"impid": "oneImp",
+			"price": 2.3
+		},
+		"bidType": "banner",
+		"bidder": "openx"
+	}, {
+		"bid": {
+			"cat": ["11_sports_22"],
+			"id": "bidFour",
+			"impid": "twoImp",
+			"price": 1.64
+		},
+		"bidType": "banner",
+		"bidder": "appnexus"
+	}, {
+		"bid": {
+			"cat": ["11_sports_22"],
+			"id": "bidFive",
+			"impid": "twoImp",
+			"price": 7.64,
+			"exp": 900
+		},
+		"bidType": "banner",
+		"bidder": "rubicon"
+	}],
+	"expectedCacheables": [{
+			"Type": "json",
+			"TTLSeconds": 600,
+			"Key": "11_sports_22_abcd-efgh-ijkl-mnop"
+		}, {
+			"Type": "json",
+			"TTLSeconds": 200,
+			"Key": "33_news_44abcd-efgh-ijkl-mnop"
+		}, {
+			"Type": "json",
+			"TTLSeconds": 0,
+			"Key": "33_news_44abcd-efgh-ijkl-mnop"
+		},
+		{
+			"Type": "json",
+			"TTLSeconds": 0,
+			"Key": "11_sports_22_abcd-efgh-ijkl-mnop"
+		},
+		{
+			"Type": "json",
+			"TTLSeconds": 900,
+			"Key": "11_sports_22_abcd-efgh-ijkl-mnop"
+		}
+	],
+	"defaultTTLs": {
+		"banner": 300,
+		"video": 3600,
+		"audio": 1800,
+		"native": 300
+	},
+	"targetDataIncludeWinners": true,
+	"targetDataIncludeBidderKeys": true,
```

Comment:
I have several questions about this test:

(1)  I don't understand how any of the bids get cached.  If I understand correctly, doCache() will be called with "bids" set to false and "vast" set to true (ultimately coming from line 93 in auction_test.go).  Line 125 in doCache() in auction.go has this test:
			if vast && topBidPerBidder.bidType == openrtb_ext.BidTypeVideo {
so we shouldn't enter the vast caching logic since all these bids have bidType of banner.  Why / how do these bids get cached?

(2) The expected cache keys are either 11_sports_22_abcd-efgh-ijkl-mnop or 33_news_44abcd-efgh-ijkl-mnop.  Given the categories are 11_sports_22 and 33_news_44, why is there no underscore between 44 and abcd in the latter while there is an underscore between 22 and abcd in the former?

(3) Shouldn't there also be a test in this directory with targetDataIncludeBidderKeys set to false to confirm that only the winning bids get cached with multiple imps?  Basically, it would be the same test as this (if this works) but with targetDataIncludeBidderKeys set to false and only two expected cached values.  I think reducing the number of cached values is the primary change for this PR that we want to confirm, though this case with targetDataIncludeBidderKeys is also necessary.  

Also, please update the PR description as already requested by Mansi.

Reply:
1) It is true that it does not make sense to VAST cache banner ads. However a request could be multiformat that accepts banner or video. In this case, it makes sense to have vast caching turned on for the video support. The down stream client will just have to be smart enough to figure out if the ad is banner or video, and display the content appropriately.

We could ad code to vast cache only if the ad is video if we have that bit of info available. But that is a separate optimization, so I am happy to let that be its own PR.